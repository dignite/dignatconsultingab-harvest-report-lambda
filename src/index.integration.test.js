const functions = require("./");
const MockDate = require("mockdate");

describe(`hours integration test with all real modules except those in the folder __mocks__`, () => {
  test("should return meta data for time entries and relevant time entries per day", async () => {
    const event = {
      path: "/my-path",
      headers: {
        "X-Forwarded-Proto": "https",
        Host: "example.uri",
      },
      requestContext: {
        stage: "integration-test",
      },
    };

    const result = await functions.hours(event);

    expect(result).toMatchSnapshot();
  });
});

describe(`hoursCsv integration test with all real modules except those in the folder __mocks__`, () => {
  test("should return status code and attachment file name and csv containing meta and per-day entries", async () => {
    MockDate.set(new Date("2011-11-22"));
    const event = {
      path: "/my-path",
      headers: {
        "X-Forwarded-Proto": "https",
        Host: "example.uri",
      },
      requestContext: {
        stage: "integration-test",
      },
    };

    const result = await functions.hoursCsv(event);

    expect(result).toMatchSnapshot();
  });
});
