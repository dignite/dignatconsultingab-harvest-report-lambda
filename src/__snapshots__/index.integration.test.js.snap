// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hours integration test with all real modules except those in the folder __mocks__ should return meta data for time entries and relevant time entries per day 1`] = `
Object {
  "body": "{\\"meta\\":{\\"description\\":\\"*All* unbilled billable hours, and any non-billable hours logged for the current month.\\",\\"totalUnbilledHours\\":10.7,\\"totalUnbilledHoursPerWeek\\":{\\"w45\\":10.7},\\"unbilledInvoice\\":{\\"excludingVAT\\":\\"2354.00 Swedish kronor\\",\\"includingVAT\\":\\"2942.50 Swedish kronor\\"},\\"csvFile\\":\\"https://example.uri/integration-test/my-path.csv\\"},\\"timeEntriesPerDay\\":[{\\"date\\":\\"2020-11-06\\",\\"name\\":\\"Programmering\\",\\"billableHours\\":8,\\"comment\\":null},{\\"date\\":\\"2020-11-05\\",\\"name\\":\\"Programmering\\",\\"billableHours\\":2.7,\\"comment\\":null}]}",
  "statusCode": 200,
}
`;

exports[`hoursCsv integration test with all real modules except those in the folder __mocks__ should return status code and attachment file name and csv containing meta and per-day entries 1`] = `
Object {
  "body": "ï»¿\\"description\\";\\"*All* unbilled billable hours, and any non-billable hours logged for the current month.\\"
\\"totalUnbilledHours\\";10.7
\\"totalUnbilledHoursPerWeek\\";\\"{\\"\\"w45\\"\\":10.7,\\"\\"w29\\"\\":0}\\"
\\"unbilledInvoice\\";\\"{\\"\\"excludingVAT\\"\\":\\"\\"2354.00 Swedish kronor\\"\\",\\"\\"includingVAT\\"\\":\\"\\"2942.50 Swedish kronor\\"\\"}\\"
\\"csvFile\\";\\"https://example.uri/integration-test/my-path.csv\\"

\\"date\\";\\"name\\";\\"billableHours\\";\\"comment\\"
\\"2020-11-06\\";\\"Programmering\\";8;
\\"2020-11-05\\";\\"Programmering\\";2.7;
\\"2019-07-17\\";\\"Ledig\\";0;
\\"2019-07-16\\";\\"Ledig\\";0;",
  "headers": Object {
    "Content-Disposition": "attachment; filename=unbilled-hours-2011-11-22.csv",
  },
  "statusCode": 200,
}
`;
